src/pip/_internal/utils/misc.py

```diff
@@ -1081,6 +1081,25 @@ def path_to_url(path):
     return url
 
 
+def build_url_from_netloc(netloc, scheme='https'):
+    """
+    Build a full URL from a netloc.
+    """
+    if netloc.count(':') >= 2 and '[' not in netloc:
```

Comment:
This function doesn't seem correct as described. For example, a netloc could be `user:pass@host:80` but not be a bare IPv6 address.